packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/Palette.types.ts

```diff
@@ -73,7 +74,18 @@ export interface GQLInvokeSingleClickOnDiagramElementToolInput {
   toolId: string;
   startingPositionX: number;
   startingPositionY: number;
-  selectedObjectId: string | null;
+  variables: GQLToolVariable[];
+}
+export interface GQLToolVariable {
+  name: string;
+  value: string;
+  type: GQLToolVariableType;
+}
+
+export enum GQLToolVariableType {
```

Comment:
Do not use enum in TypeScript, it creates a weird result, use an union instead: 

`export type GQLToolVariableType = 'STRING' | 'OBJECT_ID' | 'OBJECT_ID_ARRAY';

Reply:
ok